proposals/Proposal: add cookies.removeAll() method.md

```diff
@@ -0,0 +1,155 @@
+##### Removal of cookies with the same url and name combination but different partitionKey, path or host-only values:
+To remove all the cookies with the same `url` and `name` a developer must first call `cookies.getAll()` to get all of the cookies associated with the values. Then use the results of that call to make individual calls to `cookies.remove()` to delete each cookie. 
+
+##### Removal of cookies associated with a topLevelSite:
+To remove all the cookies associated with a `topLevelSite`, a developer must first call `cookies.getAll({})` to retrieve all partitioned and unpartitioned cookies. Then filter the results to identify  cookies that have a value for `topLevelSite` of their `partitionKey` match the desired `topLevelSite`. Finally the developer will need to make individual calls to `cookies.remove()` to delete each cookie. 
+
+#### Use Cases
+
+##### Cookie Manager:
+
+Let's say a cookie manager extension (with host permissions) is used by users to remove their cookies. Using `cookies.removeAll()` would ensure that the cookie manager extension is removing all the cookies associated with a `topLevelSite` correctly. Instead of relying on the developer of the extension to make a call to `cookies.getAll()` followed by call(s) to `cookies.remove()`. Reducing complexity for the developers while improving performance of the extension.
+
+### Known Consumers
+
+All extensions that access and/or modify cookies with awareness of partitioned cookies, through the use of the `partitionKey`  property  in  the  `cookies` extension API.
+
+## Specification
+
+
+### Schema
+
+##### Syntax
+
+```
+let removed = await browser.cookies.removeAll(
+details // object
+)
+```
+##### Parameters
+An `object` containing information to identify the cookie(s) to remove. It contains the following properties, based on the properties recognized by `cookies.remove`:
+
+>[`name`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#name) optional:
+>
+>A  `string`  representing the name of the cookie to remove.
+>
+>[`partitionKey`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#partitionkey)  optional:
+>
+>An object representing the storage partition containing the cookie.
+>
+>[`storeId`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#storeid)  optional:
+>
+>A  `string`  representing the ID of the cookie store to find the cookie in. If unspecified, the cookie is looked for by default in the current execution context's cookie store.
+>
+>[`url`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#url)
+>
+>A  `string`  representing the URL associated with the cookie.
+>
+>A details object must contain a `url` and/or a `topLevelSite`.
+
+##### Return
+
+> `Promise<Cookie[]>`:
```

Comment:
/cc @Rob--W since I know he normally has thoughts on performance-related aspects.

The existing remove() method doesn't return a full cookie, but rather a subset of its properties: name, partitionKey, storeId, and url.  I'm curious if we should perhaps lean on that instead, which avoids serializing quite as much data -- in particular, it avoids passing all the values to the extension (which may be quite large in practice).

WDYT?

(If we go that route, we could introduce a common type for it -- `CookieSummary`, `CookieMetadata`, or similar?)